src/main/java/com/fasterxml/jackson/core/io/NumberInput.java

```diff
@@ -152,6 +152,32 @@ public static long parseLong(char[] ch, int off, int len)
         long val = parseInt(ch, off, len1) * L_BILLION;
         return val + (long) parseInt(ch, off+len1, 9);
     }
+    
+    /**
+     * Parses an unsigned long made up of exactly 19 digits.
```

Comment:
this looks wrong - the call you are replacing allows signed numbers - if needs be you can change the calling code to only call this if _numberNegative is true

Reply:
This the same approach that `com.fasterxml.jackson.core.util.TextBuffer#contentsAsInt(boolean)` uses, it calls `com.fasterxml.jackson.core.io.NumberInput#parseInt(char[], int, int)` that parses an optionally singed value unsigned by ignoring the sign and later flips the sign if it had a minus sign.